packages/components/Callout/macos/CalloutWindow.swift

```diff
@@ -26,11 +19,13 @@ class CalloutWindow: NSWindow {
 		backgroundColor = .clear
 		isMovable = false
 
-		NotificationCenter.default.addObserver(self, selector: #selector(appDidChangeActive(notification:)), name: NSApplication.didResignActiveNotification, object: nil)
+		// Dismiss the Callout if the window is no longer active.
+		NotificationCenter.default.addObserver(self, selector: #selector(dismissCallout), name: NSApplication.didResignActiveNotification, object: nil)
 
+		// Dismiss the Callout if the user touched/clicked outside the bounds of the callout.
 		mouseEventMonitor = NSEvent.addLocalMonitorForEvents(matching: .leftMouseUp, handler: { (event) -> NSEvent? in
 			if (event.window != self) {
-				self.windowLifeCycleDelegate?.didDetectHitOutsideCallout(calloutWindow: self)
+				self.dismissCallout()
```

Comment:
weak self capture just in case?

Reply:
Good catch... Looked into it and the way I had it could still cause a retain cycle. Thought those were gone in Swift 🥲